app/Policies/AchievementPolicy.php

```diff
@@ -61,6 +61,12 @@ public function create(User $user): bool
 
     public function update(User $user, Achievement $achievement): bool
     {
+        // If the user has a DEVELOPER_JUNIOR role, they need to have a claim
+        // on the game and the achievement must not be promoted to Core/Official.
+        if ($user->hasRole(Role::DEVELOPER_JUNIOR)) {
+            return !$achievement->is_published && $user->hasActiveClaimOnGameId($achievement->game->id);
+        }
+
         return $user->hasAnyRole([
             Role::GAME_HASH_MANAGER,
```

Comment:
Should a hash manager have permissions to modify achievements?

Reply:
They shouldn't - removed in latest.